haystack/preview/components/file_converters/azure.py

```diff
@@ -0,0 +1,140 @@
+        self.model_id = model_id
+        self.save_json = save_json
+        self.id_hash_keys = id_hash_keys or []
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self, paths: List[Union[str, Path]], save_json: Optional[bool] = None, id_hash_keys: Optional[List[str]] = None
+    ):
+        """
+        Convert files to Documents using Azure's Document Intelligence service.
+
+        :param paths: Paths to the files to convert.
+        :param save_json: Whether to save the JSON output of the Azure API. If not set, the value passed to the
+            constructor will be used. Default: `None`
+        :param id_hash_keys: Generate the Document ID from a custom list of strings that refer to the Document's
+            attributes. If you want to ensure you don't have duplicate Documents in your DocumentStore but texts are not
+            unique, you can pass the name of the metadata to use when building the document ID (like
+            `["text", "category"]`) to this field. In this case, the ID will be generated by using the text and the
+            content of the `category` field.
+            If not set, the id_hash_keys passed to the constructor will be used.
+            Default: `None`
+        """
+        id_hash_keys = id_hash_keys or self.id_hash_keys
+        save_json = save_json if save_json is not None else self.save_json
+
+        documents = []
+        for path in paths:
+            path = Path(path)
+            with open(path, "rb") as file:
+                poller = self.document_analysis_client.begin_analyze_document(model_id=self.model_id, document=file)
+                result = poller.result()
+
+            if save_json:
+                with open(path.with_suffix(".json"), "w") as json_file:
+                    json.dump(result.to_dict(), json_file, indent=2)
+
+            file_suffix = path.suffix
+            document = AzureOCRDocumentConverter._convert_azure_result_to_document(result, id_hash_keys, file_suffix)
+            documents.append(document)
+
+        return {"documents": documents}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(
+            self,
+            endpoint=self.endpoint,
+            credential_key=self.credential_key,
+            model_id=self.model_id,
+            save_json=self.save_json,
+            id_hash_keys=self.id_hash_keys,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "AzureOCRDocumentConverter":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
+
+    @staticmethod
+    def _convert_azure_result_to_document(result: AnalyzeResult, id_hash_keys: List[str], file_suffix: str) -> Document:
+        """
+        Convert the result of Azure OCR to a Haystack text Document.
+        """
+        if file_suffix == ".pdf":
```

Comment:
Is there a stronger way to check whether the file was a PDF? Otherwise we can leave this as it is for now

Reply:
I checked whether the API somewhere returns which filetype it just processed, but unfortunately, it doesn't provide this information. I think as of now, this should be good enough.